poet/src/author_collection.rs

```diff
@@ -0,0 +1,44 @@
+use std::collections::BTreeMap;
+
+use rhai::CustomType;
+use rhai::TypeBuilder;
+
+use crate::author::Author;
+use crate::author_basename::AuthorBasename;
+
+#[derive(Clone, Default)]
+pub struct AuthorCollection {
+    authors: BTreeMap<AuthorBasename, Author>,
+}
+
+impl AuthorCollection {
+    pub fn insert(&mut self, basename: AuthorBasename, author: Author) {
+        self.authors.insert(basename, author);
+    }
+
+    pub fn values(&self) -> impl Iterator<Item = &Author> {
+        self.authors.values()
+    }
+
+    pub fn resolve(&self, names: &[String]) -> (Vec<Author>, Vec<String>) {
```

Comment:
That is my personal thing, but I do not like to use tuples as a return type; I find them less readable than returning structs. 

Because, let's say if you do `(foo, bar) = authors.resolve(...)` you would have no idea what `foo`, `bar` is. Instead if you return some specific struct like `AuthorsResolveResult { found_authors, missing_authors } = authors.resolve(...)` it would be instantly obvious.